homeassistant/components/incomfort/__init__.py

```diff
@@ -43,7 +45,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: InComfortConfigEntry) ->
     except TimeoutError as exc:
         raise InConfortTimeout from exc
 
-    coordinator = InComfortDataCoordinator(hass, data)
+    if entry.unique_id is not None:
+        # Register discovered gateway device
+        device_registry = dr.async_get(hass)
+        device_registry.async_get_or_create(
+            config_entry_id=entry.entry_id,
+            identifiers={(DOMAIN, entry.unique_id)},
+            connections={(dr.CONNECTION_NETWORK_MAC, entry.unique_id)},
+            manufacturer="Intergas",
+            name="RFGateway",
+        )
```

Comment:
We can always register the gateway device right? Just not always with connections

Reply:
Yes, I think so.

Reply:
done